src/server/acl/acl_family.cc

```diff
@@ -567,6 +568,13 @@ void AclFamily::Init(facade::Listener* main_listener, UserRegistry* registry) {
     return;
   }
   registry_->Init();
+  config_registry.RegisterMutable("aclfile", [this](const absl::CommandLineFlag& flag) {
+    auto res = flag.TryGet<std::string>();
+    if (res.has_value()) {
+      LoadToRegistryFromFile(res.value(), false);
```

Comment:
@kostasrim  I see you used `std::optional<facade::ErrorReply>` to return from `LoadToRegistryFromFile` but you never actually use `ErrorReply` abilities when calling LoadToRegistryFromFile. The advised way to pass io errors is via `io::Result` or `std::error_code`. I think `std::error_code` would suffice here. `ErrorReply` was introduced by @dranikpg  for specific purposes and it's not intended to replace `std::error_code` or `io::Result`.

Reply:
`LoadRegistryFromFile` does not only do file IO but also parsing of the ACL file. `io::Result` only describes the class of errors for the first case (IO). The second(parsing), I would not consider it low level such that it fits the class of errors described by `std::error_code` and that's why I opted in for `ErrorReply` since I find it a little more general. 

In the context of how it's used here, I think we should report an Error if the file failed to load.

Reply:
What I am saying is that no caller today does not use the returned error message and defaults for a boolean check if error/not error. LoadToRegistryFromFile already logs any error internally so why to pass free-text error messages up the stack?

Reply:
> What I am saying is that no caller today uses the returned error message and defaults for a boolean check if error/not

That's not true because `Load` is overloaded. The one without parameters is used on DF startup and therefore there is no error message to be send back to the user (only printed to the DF log if there was an error loading the file). However, with the command `ACL LOAD` we need to report the error to the user, and the second overload of `Load()` handles this case.

```
281 void AclFamily::Load(CmdArgList args, ConnectionContext* cntx) {                                     
282   auto acl_file = absl::GetFlag(FLAGS_aclfile);                                                      
283   if (acl_file.empty()) {                                                                            
284     cntx->SendError("Dragonfly is not configured to use an ACL file.");                              
285     return;                                                                                          
286   }                                                                                                  
287                                                                                                      
288   const auto is_successfull = LoadToRegistryFromFile(acl_file, !cntx);                               
289                                                                                                      
290   if (is_successfull) {                                                                              
291     cntx->SendError(absl::StrCat("Error loading: ", acl_file, " ", is_successfull->ToSv()));         
292     return;                                                                                          
293   } 
```

Tbh `std::optional<string>` would suffice here.

Reply:
I introduced ErrorReply to avoid passing mutable ConnectionContext* into functions only to return an error